policybot/pkg/testresults/testresults.go

```diff
@@ -0,0 +1,314 @@
+		}
+		if err != nil {
+			return nil, err
+		}
+		paths = append(paths, attrs.Prefix)
+
+	}
+	return paths, nil
+}
+
+// GetTest function get all directories under the given pr in blob storage for each test suite name.
+// Client: client used to get buckets and objects.
+// PrNum: the PR number inputted.
+// Return []Tests return a slice of Tests objects.
+func (prt *PrResultTester) getTests(orgLogin string, repoName string, prNumInt int64) (map[string][]string, error) {
+	prNum := strconv.FormatInt(prNumInt, 10)
+	prefixForPr := "pr-logs/pull/" + orgLogin + "_" + repoName + "/" + prNum + "/"
+	testNames, err := prt.query(prefixForPr)
+	if err != nil {
+		return nil, err
+	}
+	testMap := map[string][]string{}
+	var runs []string
+	for _, testPref := range testNames {
+		testPrefSplit := strings.Split(testPref, "/")
+		testname := testPrefSplit[len(testPrefSplit)-2]
+		runs, err = prt.query(testPref)
+		if err != nil {
+			return nil, err
+		}
+		var runPaths []string
+		var ok bool
+		if runPaths, ok = testMap[testname]; !ok {
+			runPaths = []string{}
+		}
+		for _, runPath := range runs {
+			if runPath != "" {
+				runPaths = append(runPaths, runPath)
+			}
+		}
+		testMap[testname] = runPaths
+	}
+	return testMap, nil
+}
+
+func (prt *PrResultTester) getInformationFromFinishedFile(pref string, eachRun *store.TestResult) (*store.TestResult, error) {
+	// It is possible that the folder might not contain finished.json.
+	client := prt.Client
+	bucket := client.Bucket(prt.bucketName)
+	newObj := bucket.Object(pref + "finished.json")
+	nrdr, err := newObj.NewReader(prt.ctx)
+	var finish finished
+
+	if err != nil {
+		return eachRun, err
+	}
+
+	defer nrdr.Close()
+	finishFile, err := ioutil.ReadAll(nrdr)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(finishFile, &finish); err != nil {
+		return nil, err
+	}
+
+	passed := finish.Passed
+	result := finish.Result
+	t := finish.Timestamp
+	tm := time.Unix(t, 0)
+
+	eachRun.TestPassed = passed
+	eachRun.Result = result
+	eachRun.FinishTime = tm
+
+	return eachRun, nil
+}
+
+func (prt *PrResultTester) getInformationFromStartedFile(pref string, eachRun *store.TestResult) (*store.TestResult, error) {
+	client := prt.Client
+	bucket := client.Bucket(prt.bucketName)
+	newObj := bucket.Object(pref + "started.json")
+	nrdr, err := newObj.NewReader(prt.ctx)
+	if err != nil {
+		return eachRun, err
+	}
+
+	defer nrdr.Close()
+	startFile, nerr := ioutil.ReadAll(nrdr)
+	if nerr != nil {
+		return nil, nerr
+	}
+
+	var started started
+
+	if err := json.Unmarshal(startFile, &started); err != nil {
+		return nil, err
+	}
+	t := started.Timestamp
+	tm := time.Unix(t, 0)
+	eachRun.StartTime = tm
+
+	return eachRun, nil
+}
+
+func (prt *PrResultTester) getInformationFromCloneFile(pref string, eachRun *store.TestResult) (*store.TestResult, error) {
+	client := prt.Client
+	bucket := client.Bucket(prt.bucketName)
+	obj := bucket.Object(pref + "clone-records.json")
+	scope.Infof("read clone")
```

Comment:
Remove the two uses of scope.Infof, we don't want to leave  debugging stuff in the logs.